src/irmin-pack/inode.ml

```diff
@@ -1262,9 +1272,45 @@ struct
         in
         proof_of_concrete t.hash p
 
-      let of_proof la ~depth (proof : proof) =
-        let c = concrete_of_proof ~depth proof in
-        match of_concrete ~depth la c with Ok v -> Some v | Error _ -> None
+      let of_proof (Partial _ as la) ~depth (proof : proof) =
+        match proof with
+        | `Values vs when List.compare_length_with vs Conf.entries > 0 -> (
+            if depth <> 0 then None
```

Comment:
any reason to return `None` instead of an `assert false`?

Reply:
The return type of `of_proof` is `inode option` where `None` means that there is an error in `proof`.